### DISCUSSION TITLE
Mattermostへ通知する方法はありますか？

### DISCUSSION THREAD

**Discussion by hkato312** [growilabs/growi#8713] [Help / Questions]
Webhookはレガシー扱いになっていて無効化されていますが、有効化する手順が見当たらないです。
有効化する方法はありますか？
また、将来的にWebhookが廃止された場合、Mattermostへ通知する方法はありますか？

| item     | version |
| ---      | --- |
|GROWI     |6.2.1|
|node.js   |18.18.0|
|npm       |9.8.1|
|yarn      |1.22.19|
|Using Docker|yes/no|
|Using [growi-docker-compose][growi-docker-compose]|yes/no|

[growi-docker-compose]: https://github.com/weseek/growi-docker-compose

**Comment by yuki-takei** [comment]
> Webhookはレガシー扱いになっていて無効化されていますが、有効化する手順が見当たらないです。
> 有効化する方法はありますか？

`/admin/slack-integration` で設定されている情報のリセット(全て消す)をすれば有効化できないでしょうか？
UIからできない場合はバグだと思いますので教えてください。
応急的な対応方法としては、MongoDB の configs 内の該当データを消すことになると思います。

> また、将来的にWebhookが廃止された場合、Mattermostへ通知する方法はありますか？

`/admin/notification` で設定できる通知の種類のうち、User trigger notification については手段がなくなることになりますね…
ということは廃止しない方がいいのかもしれませんね。

**Reply by hkato312** [reply]
> /admin/slack-integration で設定されている情報のリセット(全て消す)をすれば有効化できないでしょうか？
> UIからできない場合はバグだと思いますので教えてください。

ありがとうございます。/admin/slack-integration で「全ての設定をリセット」をクリックすると確認ダイアログが表示されますが、
「リセット」をクリックしても、何も変化がありません。

下記状態でも現象が再現しました。
　インストール直後の状態(Slack連携、Slack連携(レガシー)に何も設定していない状態)で
　admin/slack-integration で Slack bot の内どれかをクリック(クリックするだけで後の手続きをしない)
　⇒ 「全ての設定をリセット」でリセットできない
